src/zenml/cli/stack_components.py

```diff
@@ -235,22 +236,126 @@ def register_stack_component_command(
                 cli_utils.error(str(e))
                 return
 
-            from zenml.stack.stack_component_class_registry import (
-                StackComponentClassRegistry,
-            )
+        from zenml.stack.registry import flavor_registry
 
-            component_class = StackComponentClassRegistry.get_class(
-                component_type=component_type, component_flavor=flavor
+        zenml_flavors = flavor_registry.get_flavors_by_type(
+            component_type=component_type
+        )
+
+        try:
```

Comment:
I just saw this now, but this is coming back to my comment on the `ComponentWrapper.to_component` method: The logic there is an exact copy to the one from here, so I think we should find a better place to implement this. IMO the flavor registry might be a good place, what do you think?